src/Altinn.Platform/Altinn.Platform.Storage/IntegrationTest/DataElementStorageTests.cs

```diff
@@ -214,19 +301,30 @@ private async Task<List<DataElement>> CreateInstanceWithData(int count)
                 string requestUri = $"{versionPrefix}/instances/{newInstance.Id}/data?dataType={dataType}";
 
                 // post the file
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _validToken);
                 HttpResponseMessage postResponse = await client.PostAsync(requestUri, jsonContent.AsJson());
 
                 DataElement dataElement = JsonConvert.DeserializeObject<DataElement>(await postResponse.Content.ReadAsStringAsync());
 
-                // update downloaded structure on data element
-                dataElement.AppOwner ??= new ApplicationOwnerDataState();
-                dataElement.AppOwner.Downloaded = new List<DateTime>();
-                dataElement.AppOwner.Downloaded.Add(DateTime.UtcNow);
+                if (setDownload)
+                {
+                    // update downloaded structure on data element
+                    dataElement.AppOwner ??= new ApplicationOwnerDataState();
+                    dataElement.AppOwner.Downloaded = new List<DateTime>
+                    {
+                        DateTime.UtcNow
+                    };
 
-                requestUri = $"{versionPrefix}/instances/{newInstance.Id}/dataelements/{dataElement.Id}";
-                HttpResponseMessage putResponse = await client.PutAsync(requestUri, dataElement.AsJson());
+                    requestUri = $"{versionPrefix}/instances/{newInstance.Id}/dataelements/{dataElement.Id}";
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _validToken);
+                    HttpResponseMessage putResponse = await client.PutAsync(requestUri, dataElement.AsJson());
 
-                dataElements.Add(JsonConvert.DeserializeObject<DataElement>(await putResponse.Content.ReadAsStringAsync()));
+                    dataElements.Add(JsonConvert.DeserializeObject<DataElement>(await putResponse.Content.ReadAsStringAsync()));
```

Comment:
Consider using this instead:
Instance instance = await getResponse.Content.ReadAsAsync<DataElement>();